samples/hssi/hssi_100g_cmd.h

```diff
@@ -113,6 +141,212 @@ class hssi_100g_cmd : public hssi_cmd
+  {
+    hafu->mbox_write(CSR_SRC_ADDR_LO, static_cast<uint32_t>(bin_src_addr));
+    hafu->mbox_write(CSR_SRC_ADDR_HI, static_cast<uint32_t>(bin_src_addr >> 32));
+
+    hafu->mbox_write(CSR_DST_ADDR_LO, static_cast<uint32_t>(bin_dest_addr));
+    hafu->mbox_write(CSR_DST_ADDR_HI, static_cast<uint32_t>(bin_dest_addr >> 32));
+  }
+
+  std::ostream & print_monitor_headers(std::ostream &os) const
+  {
+    os<<"\r\n"<<"Monitor mode, Press 'q' to quit and 'r' to reset"<<"\r";
+    os<< "\r\n_________________________________________________________________________________________________\r\n";
+
+    os<< "\r\n"
+    << std::left
+    << "| "
+    << std::setw(5)
+    << "Port" << " | " << std::setw(15)
+    << "Tx count" << " | " << std::setw(15)
+    << "Rx count" << " | " << std::setw(15)
+    << "Rx good"  << " | " << std::setw(10)
+    << "Rx pkt/s" << " | " << std::setw(3)
+    << "BW" << "    " << std::setw(5) << " | "
+    << "\r";
+
+    os<< "\n---------------------------------------------------------------------------------------------------\r\n";
+    return os;
+  }
+
+  std::ostream & print_monitor_data(std::ostream &os, perf_data *data, int port) const
+  {
+    os << std::dec
+    << std::left
+    << "| "
+    << std::setw(5)
+    << port << " | " << std::setw(15)
+    << data->tx_count << " | " << std::setw(15)
+    << data->rx_count << " | " << std::setw(15)
+    << data->rx_good_packet_count<< " | " << std::setw(10)
+    << data->rx_pkt_sec << " | " << std::setw(3)
+    << data->bw << "Gbps" << std::setw(5) << " | "
+    << "\r";
+    return os;
+  }
+
+  void select_port(int port, ctrl_config config_data, hssi_afu *hafu) const
+  {
+    /* Selects the port before performing read/write to traffic controller reg space */
+    hafu->write64(TRAFFIC_CTRL_PORT_SEL, port);
+    write_ctrl_config(hafu, config_data);
+    write_csr_addr(hafu, mac_bits_for(src_addr_), mac_bits_for(dest_addr_));
+  }
+
+  void capture_perf(perf_data *old_perf_data, hssi_afu *hafu) const
+  {
+    memset(old_perf_data, 0, sizeof(perf_data));
+    if (stall_enable(hafu))
+    {
+      read_performance(old_perf_data, hafu);
+      stall_disable(hafu);
+    }
+  }
+
+  std::ostream & run_monitor(std::ostream &os, perf_data *old_perf_data, int port, hssi_afu *hafu)
+  {
+    perf_data *cur_perf_data = (perf_data*)malloc(sizeof(perf_data));
+    perf_data *new_perf_data = (perf_data*)malloc(sizeof(perf_data));
+
+    memset(cur_perf_data, 0, sizeof(perf_data));
+
+    capture_perf(new_perf_data, hafu);
+    calc_performance(old_perf_data, new_perf_data, cur_perf_data, start_size_);
+
+    print_monitor_data(os, cur_perf_data, port);
+    memcpy(old_perf_data, new_perf_data, sizeof(perf_data));
+
+    return os;
+  }
+
+  char get_user_input(char key) const
+  {
+    /* Waits for user input for 1 second */
+    key = '\0';      // initialized to null
+    std::thread t1([&](){
+      std::cin>>key;
+    });
+    t1.detach();
```

Comment:
Why detach()?  Can we not join() it?

Reply:
The child thread we have created to get user input should run independently along with main thread,
join() will wait until the child thread gets executed. hence we have used detach().
Otherwise, std::cin operation makes main program to wait until input is provided.